### ISSUE TITLE
Usage with jest unstable

### ISSUE THREAD

**Issue by derweili** [boazpoolman/strapi-plugin-config-sync#17]
I tried to used this plugin together with Jest.

I thought I could setup the user permissions using strapi admin, export the settings using this plugin and load those settings during the test runs with jest. So I do not have to manually setup the permissions during each test run.

This first worked for me, but I then noticed errors: 
Sometimes, the settings do not get imported.

Example:
When running `npm test` 5 times, it will likely work 4 times but 1 time it will fail, without anything changed in between.
In my example I have some routes wich should only be accessible for authorized users. Most of the time the requests work but sometimes I get forbidden errors.

https://user-images.githubusercontent.com/6911678/134194098-9aa14a68-f913-49bb-ad5a-a9922b94b470.mov

I suspect that the import runs asynchronously in the background and sometimes the import is not completed before the tests run. But I am not sure about that.

Is there a reference implementation for use with Jest?

I setup Jest as Described in the strapi docs: 
https://strapi.io/documentation/developer-docs/latest/guides/unit-testing.html#testing-basic-endpoint-controller

this is my config-sync config:
```js
module.exports = ({ env }) => ({
  'config-sync': {
    destination: "extensions/config-sync/files/",
    minify: false,
    importOnBootstrap: true,
    include: [
      "core-store",
      "role-permissions"
    ],
    exclude: [
      "core-store.plugin_users-permissions_grant"
    ]
  },
});
```

This is the test that's running in my example video:

```js
  it('should create ticket', async () => {
    await request(strapi.server) // app server is an instance of Class: http.Server
      .post('/tickets')
      .set('accept', 'application/json')
      .set('Content-Type', 'application/json')
      .set('Authorization', 'Bearer ' + jwt)
      .send({
        ...ticketData
      })
      .expect('Content-Type', /json/)
      .expect(200)
      .then(data => {
        expect(data.body).toBeDefined();
        expect(data.body.title).toBe(ticketData.title);
      })
  });

```

**Comment by boazpoolman** [comment]
Hi @derweili!
What an interesting usecase for this plugin. Haven't used the plugin like this myself yet, but it seems like it fits your purpose great!

The `importOnBootstrap` setting will simply run the `importAllConfig` service, just as it would when importing from the admin panel. This service is asynchronous, but is used with `await` so should be OK.

Though I did find that inside this service the plugin will rerun `importSingleConfig` for each config, though this is also a asynchronous method which is not used with `await`. Maby this fixes the issue?

Are you in the position to test this with your setup as described above? If so I could create a PR for you to test.

**Comment by derweili** [comment]
@boazpoolman yes I can test it again. What do I have to do?

Install a newer version of this plugin?

**Comment by derweili** [comment]
@boazpoolman I just testet the PR, unfortunately I still get the error.

I also sometimes (very rarely) get an error when initiating the Strapi instance. But because this only happend two times, I wasn't able to verify if this is related to this plugin. 

Don't you have to return this await importSingleConfig function?
https://github.com/boazpoolman/strapi-plugin-config-sync/pull/23/files#diff-e487fde79a6fb1723e2b7bcb2fd3e62ba4148f537b0d9cd725ecacd2bc6f57a0R160
Although this also does not solve the error, I tried this already.

**Comment by boazpoolman** [comment]
Strapi v4 has been released, and this plugin was migrated along with it.

In the migrated plugin this should likely be fixed as a lot of improvements have been made for the `importOnBootstrap` setting.
Alternatively you could use the CLI to import the config before you run your test command:
```
# Using yarn
yarn cs import -y && [test-command]

# Using npm
npm run cs import -y && [test-command]
```

Released in [v1.0.0-alpha.1](https://github.com/boazpoolman/strapi-plugin-config-sync/releases/tag/1.0.0-alpha.1) of `strapi-plugin-config-sync`.
